encode-decode/JwtDecode.js

```diff
@@ -0,0 +1,42 @@
+// JWT Decode by 0mgfriday
+var Base64 = Java.type("java.util.Base64")
+var String = Java.type("java.lang.String")
+
+/**
+ * Decode JWT into JSON
+ *
+ * @param {EncodeDecodeScriptHelper} helper - A helper object with various utility methods.
+ *     For more details see https://github.com/zaproxy/zap-extensions/tree/main/addOns/encoder/src/main/java/org/zaproxy/addon/encoder/processors/script/EncodeDecodeScriptHelper.java
+ * @param {String} value - JWT to decode
+ * @returns {EncodeDecodeResult} - Decoded JWT (JSON)
+ */
+function process(helper, value){
+    const parts = value.split('.')
+    
+    if (parts.length == 2 || parts.length == 3) {
+        try {
+            let result = formatJson(b64decode(parts[0])) + '\n' +  formatJson(b64decode(parts[1]))
+        
+            if (parts.length == 3 && parts[2] != '') {
+                result += '\n{SIGNATURE}'
+            }
+
+            return helper.newResult(result);
+        } catch (err) {
+            return helper.newError("Invalid JWT: Unable to decode");
+        }
+    }
+
+    return helper.newError("Invalid JWT");
+}
+
+function b64decode(s) {
+    s = s.replace('-', '+')
+    s = s.replace('_', '/')
+    var bytes = Base64.getDecoder().decode(s)
```

Comment:
Use the URL variant instead of manual replacements.

Reply:
updated